inductiva/resources/machine_types.py

```diff
@@ -1,19 +1,75 @@
 """Available machine types and their number of cores."""
-AVAILABLE_MACHINES = {
-    "c2-standard-": [4, 8, 16, 30, 60],
-    "c3-standard-": [4, 8, 22, 44, 88, 176],
-    "c2d-standard-": [2, 4, 8, 16, 32, 56, 112],
-    "c2d-highcpu-": [2, 4, 8, 16, 32, 56, 112],
-    "e2-standard-": [2, 4, 8, 16, 32],
-    "n2-standard-": [2, 4, 8, 16, 32, 48, 64, 80, 96, 128],
-    "n2d-standard-": [2, 4, 8, 16, 32, 48, 64, 80, 96, 128, 224],
-    "n1-standard-": [1, 2, 4, 8, 16, 32, 64, 96],
-    "e2-highcpu-": [2, 4, 8, 16, 32],
-}
 
 
 def list_available_machines():
-    """Lists the types of available machines."""
-    return (machine_type + str(vcpu)
-            for machine_type, vcpus in AVAILABLE_MACHINES.items()
-            for vcpu in vcpus)
+    """Lists the available machines from the descriptive dict.
+    
+    Each machine type is a key in the dict with respective possible values are:
+    - vcpus: list of integers with the available vcpu possibilities.
+    - memory: list of string with the available memory types. Default is
+        ["highmem", "standard", "highcpu"].
+    - extra_configs: tuple with two lists of strings and integers. The first
+        list is extra memory types that are the only ones containing the
+        possible vcpus in the second list.
+    - lssd: boolean indicating if the machine type has the local SSD
+        option. Default is False.
+    """
+
+    available_machines = []
+
+    for machine_type, configs in AVAILABLE_MACHINES.items():
+        vcpu_list = configs["vcpus"]
+        memory_list = configs.get("memory", ["highmem", "standard", "highcpu"])
+        lssd = configs.get("lssd", False)
+        extra_configs = configs.get("extra_configs", None)
+
+        for memory in memory_list:
+            for vcpu in vcpu_list:
+                available_machines.append(machine_type + "-" + memory + "-" +
+                                          str(vcpu))
+
+        if extra_configs:
+            for memory in extra_configs[0]:
+                for vcpu in extra_configs[1]:
+                    available_machines.append(machine_type + "-" + memory +
+                                              "-" + str(vcpu))
+        if lssd:
+            for vcpu in vcpu_list[1:]:
+                available_machines.append(machine_type + "-standard-" +
+                                          str(vcpu) + "-lssd")
+
+    return tuple(available_machines)
+
+
+AVAILABLE_MACHINES = {
+    "c2": {
+        "vcpus": [4, 8, 16, 30, 60],
+        "memory": ["standard"]
+    },
+    "c3": {
+        "vcpus": [4, 8, 22, 44, 88, 176],
+        "lssd": True
+    },
```

Comment:
this is what I get when I run the code in this file for the `c3` machines:

```
c3-highmem-4
c3-highmem-8
c3-highmem-22
c3-highmem-44
c3-highmem-88
c3-highmem-176
c3-standard-4
c3-standard-8
c3-standard-22
c3-standard-44
c3-standard-88
c3-standard-176
c3-highcpu-4
c3-highcpu-8
c3-highcpu-22
c3-highcpu-44
c3-highcpu-88
c3-highcpu-176
c3-standard-8-lssd
c3-standard-22-lssd
c3-standard-44-lssd
c3-standard-88-lssd
c3-standard-176-lssd
```

Is this correct? I would suppose that `lssd` would apply to all 🤔 !